src/main/java/hudson/plugins/emailext/ExtendedEmailPublisherDescriptor.java

```diff
@@ -577,23 +583,36 @@ public boolean configure(StaplerRequest req, JSONObject formData)
             throws FormException {
 
         // Configure the smtp server
-        smtpHost = nullify(req.getParameter("ext_mailer_smtp_server"));
+        mailAccount.setSmtp_host(nullify(req.getParameter("ext_mailer_smtp_server")));
         defaultSuffix = nullify(req.getParameter("ext_mailer_default_suffix"));
 
+        mailAccount.setAdv_properties(nullify(req.getParameter("ext_mailer_adv_properties")));
+
+        addAccounts.clear();
+        Object addacc = formData.opt("addAccounts");
+        if(addacc != null){
+            if(addacc instanceof JSONArray){
+                for(Object obj : (JSONArray)addacc) addAccounts.add(new MailAccount((JSONObject)obj));
+            }
+            else if(addacc instanceof JSONObject){
+                addAccounts.add(new MailAccount((JSONObject)addacc));
+            }
+        }
+
         // specify authentication information
         if (req.hasParameter("ext_mailer_use_smtp_auth")) {
-            smtpAuthUsername = nullify(req.getParameter("ext_mailer_smtp_username"));
-            smtpAuthPassword = Secret.fromString(nullify(req.getParameter("ext_mailer_smtp_password")));
+            mailAccount.setSmtp_username(nullify(req.getParameter("ext_mailer_smtp_username")));
+            mailAccount.setSmtp_password(nullify(req.getParameter("ext_mailer_smtp_password")));
```

Comment:
Why the loss of Secret.fromString

Reply:
because it makes inside of class

Reply:
That is by design and is required to store the password encrypted

Reply:
oh sorry missed it in the subclass